lib/string/strdup/strndup.c

```diff
@@ -0,0 +1,7 @@
+// SPDX-FileCopyrightText: 2025, Alejandro Colomar <alx@kernel.org>
+// SPDX-License-Identifier: BSD-3-Clause
+
+
+#include <config.h>
+
+#include "string/strdup/strndup.h"
```

Comment:
> This translation unit helps verify that the header file is self-contained. I tend to always add one translation unit per header file.

There are unit test for exactly that purpose :-)
It does not make much benefit to include empty module to library  for use-case which should be covered by tests.

Or it should be documented in the file why such unnecessary module exist. Just to be sure it will not be removed in the future as part of cleanup.